src/index.ts

```diff
@@ -0,0 +1,185 @@
+import {
+  getInput,
+  getMultilineInput,
+  info,
+  setFailed,
+  warning,
+  endGroup,
+  startGroup,
+} from "@actions/core";
+import { execSync, spawnSync } from "node:child_process";
+import { existsSync } from "node:fs";
+import * as path from "node:path";
+
+const config = {
+  WRANGLER_VERSION:
+    getInput("wranglerVersion") !== "" ? getInput("wranglerVersion") : "latest",
+  secrets: getMultilineInput("secrets"),
+  workingDirectory: checkWorkingDirectory(getInput("workingDirectory")),
+  CLOUDFLARE_API_TOKEN: getInput("apiToken"),
+  CLOUDFLARE_ACCOUNT_ID: getInput("accountId"),
+  ENVIRONMENT: getInput("environment"),
+  VARS: getMultilineInput("vars"),
+  COMMANDS: getMultilineInput("commands"),
+};
+
+export async function main() {
+  installWrangler();
+  authenticationSetup();
+  await execCommands(getMultilineInput("preCommands"));
+  await uploadSecrets();
+  await genericCommand();
+  await execCommands(getMultilineInput("postCommands"));
+}
+
+function checkWorkingDirectory(workingDirectory = "") {
+  try {
+    const normalizedPath = path.normalize(workingDirectory);
+    if (existsSync(normalizedPath)) {
+      return normalizedPath;
+    } else {
+      setFailed(`🚨 Directory ${workingDirectory} does not exist.`);
+    }
+  } catch (error) {
+    setFailed(
+      `🚨 While checking/creating directory ${workingDirectory} received ${error}`
+    );
+  }
+}
+
+function installWrangler() {
+  startGroup("📥 Installing Wrangler");
+  const command = `npx install wrangler@${config["WRANGLER_VERSION"]}`;
+  info(`Running Command: ${command}`);
+  spawnSync(command, { cwd: config["workingDirectory"], env: process.env });
+  endGroup();
+}
+
+function authenticationSetup() {
+  startGroup("🔐 Authenticating with Cloudflare");
+  try {
+    process.env.CLOUDFLARE_API_TOKEN = config["CLOUDFLARE_ACCOUNT_ID"];
+    process.env.CLOUDFLARE_ACCOUNT_ID = config["CLOUDFLARE_API_TOKEN"];
+    info(`Authentication process initiated with - API Token`);
+  } catch (error) {
+    setFailed(
+      `Authentication details were not found. Please input an 'apiToken' to the action.`
+    );
+  }
+  endGroup();
+}
+
+async function execCommands(commands: string[]) {
+  startGroup("🚀 Executing Pre/Post Commands");
+  if (!commands.length) {
+    warning(`📌 Pre/Post Commands were not provided, skipping execution.`);
+    return;
+  }
+  for (const command of commands) {
+    const npxCMD = command.startsWith("wrangler")
+      ? `npx ${command}`
+      : command;
+
+    info(`🚀 Executing command: ${npxCMD}`);
+
+    execSync(npxCMD, {
+      cwd: config["workingDirectory"],
+      env: process.env,
+    });
+  }
+  endGroup();
+}
+
+async function uploadSecrets() {
+  startGroup("🔑 Uploading Secrets");
+  const secrets: string[] | string = config["secrets"];
+  if (!secrets.length) {
+    warning(`📌 No secrets were provided, skipping upload.`);
+    return;
+  }
```

Comment:
Those are groups, which actually cleanup the console significantly rather than them all logging at the top-level. Could name it better so it doesnt sound like its actively doing it.

Reply:
Yeah, but the `groupStart()` call still logs a heading immediately upon being called. I'm all for grouping log messages to make them easier to visually parse. My suggestion is that users who aren't using any secrets shouldn't see log messages about secrets.

Reply:
That's reasonable.